ROMFS/px4fmu_common/init.d/rc.mavlink

```diff
@@ -82,132 +84,135 @@ fi
 #
 # XXX We need a better way for runtime eval of shell variables,
 # but this works for now
-if param compare SYS_COMPANION 10
+if [ "x${MAVLINK_COMPANION_DEVICE}" != xnone ]
 then
-	frsky_telemetry start -d ${MAVLINK_COMPANION_DEVICE}
-else
-	if ver hwcmp PX4FMU_V4 PX4FMU_V4PRO MINDPX_V2
+	if param compare SYS_COMPANION 10
 	then
-		# This is TELEM4 on Pixhawk 3 Pro
-		frsky_telemetry start -d /dev/ttyS6 -t 15
+		frsky_telemetry start -d ${MAVLINK_COMPANION_DEVICE}
+	else
+		if ver hwcmp PX4FMU_V4 PX4FMU_V4PRO MINDPX_V2
+		then
+			# This is TELEM4 on Pixhawk 3 Pro
+			frsky_telemetry start -d /dev/ttyS6 -t 15
+		fi
 	fi
-fi
 
-if param compare SYS_COMPANION 20
-then
-	syslink start
-	mavlink start -d /dev/bridge0 -b 57600 -m osd -r 40000
-fi
+	if param compare SYS_COMPANION 20
+	then
+		syslink start
+		mavlink start -d /dev/bridge0 -b 57600 -m osd -r 40000
```

Comment:
This one isn't dependant on MAVLINK_COMPANION_DEVICE.

Reply:
I do not follow the intent of this part of the RC well. 

Do we need to discriminate? - The default is not 'none'  - how many knobs should we have and what do they do? 

SYS_COMPANION is a 'select one mode' Knob.
MAVLINK_COMPANION_DEVICE is the Knob that selects the device
Do we add HAS_MAVLINK_COMPANION?

Options:
param set SYS_COMPANION -1 in NXPhlite
Leave MAVLINK_COMPANION_DEVICE as none and let mavling fail to open it. (does mavlink fail well?)

What else?

How do you see this best resolved?

Reply:
@dagar - I split them out into dependant on MAVLINK_COMPANION_DEVICE